docs/txRelay.md

```diff
@@ -0,0 +1,44 @@
+# TxRelay
+TxRelay is a contract that provides the uPort contract system with meta transactions. Meta-tx are a way for a user to sign some data, and then another person/service to relay this data to the Ethereum network. This has the benefit of allowing “unfunded” keys to exist and transact on the Ethereum network without the difficulty of funding them. Instead the person/service relaying the transaction will pay for the gas cost.
+
+## Requirements
+* User should be able to send tx without having ether to pay for gas
+* Relayer can be chosen by the user
+* Minimal overhead
+* Similar security to regular transaction
+    * Atomic
+    * As non-replayable as possible
+
+## Design
+The TxRelay contract has one main function, called `relayMetaTx`, that takes in a signature, destination, and data as parameters. This function checks that the signature is valid, then calls the destination address with the data parameter. The destination address must be a contract that is compatible with the TxRelay contract. The relayer can be chosen by the user and is part of the signature described below.
+
+### Signed data
+Currently, a user is required to sign the following data: relay_address :: nonce :: destination :: data :: relayer_address
+
+Rationale:
+1. relay_address: if the txRelay is ever upgraded, these transactions cannot be replayed.
+2. nonce: prevents replay attacks. 
+3. destination: necessary to know where tx is going :)
+4. data: necessary to know what the user is trying to do. 
+5. relayer_address: prevents a not-really-feasible-but-maybe attack, where someone could front run, but not send enough gas when they do for the meta-tx to be processed, and then try and trick the user to convince them the transaction just failed, other than the meta-tx. They can then relay the transaction later. Assuming an honest relayer, this is not an issue. 
```

Comment:
Oh, that's why you can specify gas for a subcall, so that you can have enough gas in the first call to be able to finish the tx if the subcall runs oog.